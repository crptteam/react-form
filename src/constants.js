import React from 'react';
import { isEqual } from 'lodash';


export const FormContext = React.createContext();

export const withConsumer = Component => {
  return class extends React.Component {
    shouldComponentUpdate(nextProps, nextState) {
      return !isEqual(nextProps, this.props) || !isEqual(nextState, this.state);
    }

    render() {
      return (
        <FormContext.Consumer>
          {({ curryOnChange, onFieldMount }) => (
            <Component
              curryOnChange={curryOnChange}
              onFieldMount={onFieldMount}
              {...this.props}
            />
          )}
        </FormContext.Consumer>
      );
    }
  };
};
