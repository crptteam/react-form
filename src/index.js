import Form from './components/Form';
import FormLine from './components/FormLine';
import CollapsibleContent from './components/CollapsibleContent';

export { Form, FormLine, CollapsibleContent };
