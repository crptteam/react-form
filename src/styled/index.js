export { default as FormWrapper } from './FormWrapper';
export { default as FieldWrapper } from './FieldWrapper';
