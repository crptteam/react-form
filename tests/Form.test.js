import React from "react";
import { Form, FormLine, CollapsibleContent } from "../src/";

describe("Form", () => {
  it("Should renders without problems", () => {
    const wrapper = shallow(<Form>
      <CollapsibleContent>
        <FormLine>I am form.</FormLine>
      </CollapsibleContent>
    </Form>);

    expect(wrapper).toMatchSnapshot();
  });
});
