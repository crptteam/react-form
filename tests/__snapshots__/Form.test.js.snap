// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Should renders without problems 1`] = `
<FormWrap
  innerRef={[Function]}
  onSubmit={[Function]}
>
  <CollapsibleContent
    key=".0"
    visible={false}
  >
    <FormLine
      key=".0"
    >
      I am form.
    </FormLine>
  </CollapsibleContent>
</FormWrap>
`;
